mirgecom/viscous.py

```diff
@@ -0,0 +1,377 @@
+        Fluid state
+    grad_t: numpy.ndarray
+        Gradient of the fluid temperature
+
+    Returns
+    -------
+    numpy.ndarray
+        The conductive heat flux vector
+    """
+    transport = eos.transport_model()
+    return -transport.thermal_conductivity(eos, cv)*grad_t
+
+
+def diffusive_heat_flux(discr, eos, cv, j):
+    r"""Compute the diffusive heat flux, ($\mathbf{q}_{d}$).
+
+    The diffusive heat flux is defined by:
+
+    .. math::
+
+        \mathbf{q}_{d} = \sum_{\alpha=1}^{\mathtt{Nspecies}}{h}_{\alpha}
+        \mathbf{J}_{\alpha},
+
+    with species specific enthalpy ${h}_{\alpha}$ and diffusive flux
+    ($\mathbf{J}_{\alpha}$) defined as:
+
+    .. math::
+
+        \mathbf{J}_{\alpha} = -\rho{d}_{\alpha}\nabla{Y}_{\alpha},
+
+    where ${Y}_{\alpha}$ is the vector of species mass fractions.
+
+    Parameters
+    ----------
+    eos: mirgecom.eos.GasEOS
+        A gas equation of state with a non-empty
+        :class:`~mirgecom.transport.TransportModel`
+    cv: :class:`~mirgecom.fluid.ConservedVars`
+        Fluid state
+    j: numpy.ndarray
+        The species diffusive flux vector
+
+    Returns
+    -------
+    numpy.ndarray
+        The total diffusive heath flux vector
+    """
+    nspec = cv.nspecies
+    if nspec > 0:
+        try:
+            h_alpha = eos.get_species_enthalpies(cv)
+            return sum(h_alpha[i]*j[i] for i in range(nspec))
+        except NotImplementedError:
+            return 0
+    return 0
+
+
+# TODO: We could easily make this more general (dv, grad_dv)
+# so that user can define arbitrary array containers for *dv*,
+# enabling support for other models and simultaneous eqns.
+def viscous_flux(discr, eos, cv, grad_cv, grad_t):
+    r"""Compute the viscous flux vectors.
+
+    The viscous fluxes are:
+
+    .. math::
+
+        \mathbf{F}_V = [0,\tau\cdot\mathbf{v} - \mathbf{q},
+        \tau_{:i},-\mathbf{J}_\alpha],
+
+    with fluid velocity ($\mathbf{v}$), viscous stress tensor
+    ($\mathbf{\tau}$), and diffusive flux for each species
+    ($\mathbf{J}_\alpha$).
+
+    .. note::
+
+        The fluxes are returned as a :class:`mirgecom.fluid.ConservedVars`
+        object with a *dim-vector* for each conservation equation. See
+        :class:`mirgecom.fluid.ConservedVars` for more information about
+        how the fluxes are represented.
+
+    Parameters
+    ----------
+    eos: :class:`~mirgecom.eos.GasEOS`
+        A gas equation of state with a non-empty
+        :class:`~mirgecom.transport.TransportModel`
+    cv: :class:`~mirgecom.fluid.ConservedVars`
+        Fluid state
+    grad_cv: :class:`~mirgecom.fluid.ConservedVars`
+        Gradient of the fluid state
+    grad_t: numpy.ndarray
+        Gradient of the fluid temperature
+
+    Returns
+    -------
+    :class:`~mirgecom.fluid.ConservedVars`
+        The viscous transport flux vector
+    """
+    transport = eos.transport_model()
+    if transport is None:
+        return 0
```

Comment:
The intent is that if there are no viscous transport properties provided, then the viscous fluxes are zero.

Reply:
Ah ok. I think this is fine then, now that the transport model language has been weakened in the docstring. Maybe also consider returning a zero `ConservedVars` to match the documentation?

Reply:
Please see what you think of this (cfe216d), and if you still think it would be better to construct a ConservedVars object full of zeros, then I'll switch it up.